packages/instantsearch-ui-components/src/components/Hits.tsx

```diff
@@ -1,9 +1,10 @@
 /** @jsx createElement */
 import { cx } from '../lib';
 
+// These will need to be imported from a shared package in the future
```

Comment:
```mermaid
flowchart TD
  types --> instantsearch-core
  connectors --> instantsearch-core
  index --> instantsearch-core
  instantsearch --> instantsearch-core
  routers? --> instantsearch-core
  components --> js
  components --> react
  components --> vue
  instantsearch-core --> components
  instantsearch-core --> js
  instantsearch-core --> react
  instantsearch-core --> vue
```

This is probably the right order indeed. We can start with moving only types, then move connectors + other things after in a non-breaking way (although we need to make sure to re-export the types from the place they are right now, otherwise it's a BC)

Reply:
I've pushed an alternative with alternative simplified types that don't require to be imported from instantsearch.js. In Hits' case I don't see an issue, as they're not exposed to contributors.

Reply:
Yes, as we're not calling it particularly much, it's ok, "components" accepts a lot, but we have precise types in the usage 🐐